src/pages/resultsView/download/DownloadTab.tsx

```diff
@@ -359,11 +403,12 @@ export default class DownloadTab extends React.Component<IDownloadTabProps, {}>
                                 <tbody>
                                     {hasValidData(this.cnaData.result!) && this.cnaDownloadControls()}
                                     {hasValidMutationData(this.mutationData.result!) && this.mutationDownloadControls()}
-                                    {hasValidData(this.mrnaData.result!) && this.mrnaExprDownloadControls()}
-                                    {hasValidData(this.proteinData.result!) && this.proteinExprDownloadControls()}
+                                    {hasValidData(this.mrnaData.result!) && this.mrnaExprDownloadControls(this.props.store.selectedMolecularProfiles.result!.find((profile) => profile.molecularAlterationType === AlterationTypeConstants.MRNA_EXPRESSION)!.name)}
+                                    {hasValidData(this.proteinData.result!) && this.proteinExprDownloadControls(this.props.store.selectedMolecularProfiles.result!.find((profile) => profile.molecularAlterationType === AlterationTypeConstants.PROTEIN_LEVEL)!.name)}
```

Comment:
Once we have valid mRNA/protein data, we will only get one molecular profile in the selectedMolecularProfiles that matches the specific molecularAlterationType. We pass the profile name to the download controls.